cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -82,23 +88,81 @@ pub struct CheckedBuildMetadata {
     _private: (),
 }
 
+#[allow(async_fn_in_trait)]
+pub trait BinaryCache {
+    async fn upload(&self, path: &str) -> Result<(), PublishError>;
+    fn cache_uri(&self) -> &Url;
+}
+
+pub struct NixCopyCache {
+    pub uri: Url,
```

Comment:
on the nix side they made an effort a few years back to change uri -> url, and since address them as urls only. It might be debatable whether they actually use the uri superset of urls or not, but for consistency with the underlying impl i'd stick with ur**L**.